test/ArgTools.jl.faceup

```diff
@@ -0,0 +1,338 @@
+        rethrow()
+    «k:end»
+    close(io)
+    «k:return» file
+«k:end»
+
+«k:function» «f:arg_write»(f::«t:Function», arg::«t:IO»)
+    «k:try» f(arg)
+    «k:finally»
+        flush(arg)
+    «k:end»
+    «k:return» arg
+«k:end»
+
+«s:"""
+    arg_isdir(f::Function, arg::AbstractString) -> f(arg)
+
+The `arg_isdir` function takes `arg` which must be the path to an existing
+directory (an error is raised otherwise) and passes that path to `f` finally
+returning the result of `f(arg)`. This is definitely the least useful tool
+offered by `ArgTools` and mostly exists for symmetry with `arg_mkdir` and to
+give consistent error messages.
+"""»
+«k:function» «f:arg_isdir»(f::«t:Function», arg::«t:AbstractString»)
+    isdir(arg) || error(«s:"arg_isdir: »«:julia-ts-string-interpolation-face:$(»«D:repr(arg)»«:julia-ts-string-interpolation-face:)»«s: not a directory"»)
+    «k:return» f(arg)
+«k:end»
+
+«s:"""
+    arg_mkdir(f::Function, arg::AbstractString) -> arg
+    arg_mkdir(f::Function, arg::Nothing) -> mktempdir()
+
+The `arg_mkdir` function takes `arg` which must either be one of:
+
+- a path to an already existing empty directory,
+- a non-existent path which can be created as a directory, or
+- `nothing` in which case a temporary directory is created.
+
+In all cases the path to the directory is returned. If an error occurs during
+`f(arg)`, the directory is returned to its original state: if it already existed
+but was empty, it will be emptied; if it did not exist it will be deleted.
+"""»
+«k:function» «f:arg_mkdir»(f::«t:Function», arg::«t:Union»{«t:AbstractString», «t:Nothing»})
+    «v:existed» = «c:false»
+    «k:if» arg === «b:nothing»
+        «v:arg» = mktempdir()
+    «k:else»
+        «v:st» = stat(arg)
+        «k:if» !ispath(st)
+            mkdir(arg)
+        «k:elseif» !isdir(st)
+            error(«s:"arg_mkdir: »«:julia-ts-string-interpolation-face:$(»«D:repr(arg)»«:julia-ts-string-interpolation-face:)»«s: not a directory"»)
+        «k:else»
+            isempty(readdir(arg)) ||
+                error(«s:"arg_mkdir: »«:julia-ts-string-interpolation-face:$(»«D:repr(arg)»«:julia-ts-string-interpolation-face:)»«s: directory not empty"»)
+            «v:existed» = «c:true»
+        «k:end»
+    «k:end»
+    «k:try» f(arg)
+    «k:catch»
+        «k:if» existed
+            «k:for» name «k:in» readdir(arg)
+                «v:path» = joinpath(arg, name)
+                prepare_for_deletion(path)
+                rm(path, «:julia-ts-quoted-symbol-face:force»=«c:true», «:julia-ts-quoted-symbol-face:recursive»=«c:true»)
+            «k:end»
+        «k:else»
+            prepare_for_deletion(arg)
+            rm(arg, «:julia-ts-quoted-symbol-face:force»=«c:true», «:julia-ts-quoted-symbol-face:recursive»=«c:true»)
+        «k:end»
+        rethrow()
+    «k:end»
+    «k:return» arg
+«k:end»
+
+«x:## test utilities ##»
+
+«k:const» «v:ARG_READERS» = [
+    String      => path -> f -> f(path)
```

Comment:
Would be nice if `->` was fontified as keyword, but julia-mode doesn't currently do this, and it is unrelated to this PR.

Reply:
It is fontified as a keyword if you crank up the fontification `treesit-font-lock-level` to 4.

Reply:
When I crank `treesit-font-lock-level` up to 4, it looks like this gets `font-lock-type-face` which isn't correct. Looks like this happens for all the punctuation: "=", "=>", "||", etc.

Reply:
Yes, again, this was the choice of the original package developers. It can be changed easily, though.